cli/tests/program.rs

```diff
@@ -1614,3 +1825,176 @@ fn test_cli_program_dump() {
         assert_eq!(program_data[i], out_data[i]);
     }
 }
+
+fn fetch_pre_signer(cmd_output_json_str: &str, target_pub_key: &str) -> Presigner {
+    let json: Value = serde_json::from_str(cmd_output_json_str).unwrap();
+    let target_pre_signer_str = json
+        .as_object()
+        .unwrap()
+        .get("signers")
+        .unwrap()
+        .as_array()
+        .unwrap()
+        .iter()
+        .find(|pre_signer| pre_signer.as_str().unwrap().starts_with(target_pub_key))
+        .unwrap()
+        .as_str()
+        .unwrap();
+    let mut parts = target_pre_signer_str.split('=');
+    assert_eq!(2, parts.clone().count());
+    Presigner::new(
+        &Pubkey::from_str(parts.next().unwrap()).unwrap(),
+        &Signature::from_str(parts.next().unwrap()).unwrap(),
+    )
+}
```

Comment:
Is it possible to use `solana_cli_output::parse_sign_only_reply_string` along with `presigner_of` instead of having this function? You can see how that's being used in `cli/tests/transfer.rs`

Reply:
Yeah, sorry for reimplementing all this stuff on my own and thanks for all these suggestions ;) Takes some familiarity with existing code to actually know what you can/can't do already,

changed in https://github.com/solana-labs/solana/pull/33860/commits/1736f75b9753d6aca90e1d0ccfae87ec59b37d31